packages/common/infra/src/sync/indexer/impl/indexeddb/index.ts

```diff
@@ -0,0 +1,124 @@
+import {
+  type AggregateOptions,
+  type AggregateResult,
+  type Document,
+  type Index,
+  type IndexStorage,
+  type IndexWriter,
+  type Query,
+  type Schema,
+  type SearchOptions,
+  type SearchResult,
+} from '../../';
+import { DataStruct, type DataStructRWTransaction } from './data-struct';
+
+export class IndexedDBIndex<S extends Schema> implements Index<S> {
+  data: DataStruct = new DataStruct(this.databaseName, this.schema);
+
+  constructor(
+    private readonly schema: S,
+    private readonly databaseName: string = 'indexer'
+  ) {}
+
+  async get(id: string): Promise<Document<S> | null> {
+    return (await this.getAll([id]))[0] ?? null;
+  }
+
+  async getAll(ids: string[]): Promise<Document<S>[]> {
+    const trx = await this.data.readonly();
+    return this.data.getAll(trx, ids);
+  }
+
+  async write(): Promise<IndexWriter<S>> {
+    return new IndexedDBIndexWriter(this.data, await this.data.readwrite());
```

Comment:
add a option to create [relaxed durability](https://developer.mozilla.org/en-US/docs/Web/API/IDBTransaction/durability#relaxed) transaction? in next pr we will create write transaction frequently, default durability strategy will require browser frush fs every time when we make a new index for doc

Reply:
In my testing, write performance is not an issue